ModelicaReference/package.mo

```diff
@@ -1902,20 +2050,18 @@ general Modelica class:
      actual arguments or default values, i.e., they may not be assigned
      values in the body of the function.</li>
 
-<li> A function may not be used in connections, may have no equations,
-     may have no initial algorithm, and can have at most one algorithm
-     section, which, if present, is the body of the function.</li>
+<li> A function may not be used in connections, may not have equations, may not have initial algorithms.</li>
 
-<li> A function may have zero or one external function interface, which,
-     if present, is the external definition of the function.</li>
+<li> A function can have at most one algorithm section or one external function interface (not both), which, if present, is the body of the function.</li>
 
-<li> For a function to be called in a simulation model, it must have
-     either an algorithm section or an external function interface as
-     its body, and it may not be partial.</li>
+<li> For a function to be called in a simulation model, the function may not be partial,
+     and the output variables must be assigned inside the function either in declaration assignments 
+	 or in an algorithm section, or have an external function interface as its body, or be defined as a function partial derivative. 
+	 The output variables of a function should be computed.</li>
```

Comment:
Should we say "All the output variables" ?

Reply:
Maybe at some time, but it is copied straight from the 3.4 specification that we describe, so I don't think we should change it here (especially since it seems minor).